QKSMS/src/main/java/com/moez/QKSMS/model/SlideshowModel.java

```diff
@@ -223,20 +246,32 @@ public static SlideshowModel createFromPduBody(Context context, PduBody pb) thro
                 } catch (UnsupportContentTypeException e) {
                     Log.e(TAG, e.getMessage(), e);
                 }
-            }
 
-            SlideModel slide = new SlideModel((int) (par.getDur() * 1000), mediaSet);
-            slide.setFill(par.getFill());
-            SmilHelper.addParElementEventListeners((EventTarget) par, slide);
-            slides.add(slide);
+                SlideModel slide = new SlideModel((int) (par.getDur() * 1000), mediaSet);
+                slide.setFill(par.getFill());
+                SmilHelper.addParElementEventListeners((EventTarget) par, slide);
+
+                if(sme.getTagName().equals(SmilHelper.ELEMENT_TAG_TEXT)){
+                    body = slide.getText();
+                    continue;
+                }
+                else{
```

Comment:
Also thank you very much for the changes, I hope we can get it to pass the tests soon.